stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -660,10 +663,27 @@ function exp!(A::StridedMatrix{T}) where T<:BlasFloat
         A2 = A * A
         A4 = A2 * A2
         A6 = A2 * A4
-        U  = A * (A6 * (CC[14].*A6 .+ CC[12].*A4 .+ CC[10].*A2) .+
-                  CC[8].*A6 .+ CC[6].*A4 .+ CC[4].*A2 .+ CC[2].*Inn)
-        V  = A6 * (CC[13].*A6 .+ CC[11].*A4 .+ CC[9].*A2) .+
-                   CC[7].*A6 .+ CC[5].*A4 .+ CC[3].*A2 .+ CC[1].*Inn
+        Ut = CC[4]*A2
+        Ut[diagind(Ut)] .+=  CC[2]
+        # Allocation economical version of:
+        #U  = A * (A6 * (CC[14].*A6 .+ CC[12].*A4 .+ CC[10].*A2) .+
+        #          CC[8].*A6 .+ CC[6].*A4 .+ Ut)
+        U = mul!(CC[8].*A6 .+ CC[6].*A4 .+ Ut,
+                 A6,
+                 CC[14].*A6 .+ CC[12].*A4 .+ CC[10].*A2,
+                 true, true)
+        U *= A
```

Comment:
This needs to be
```suggestion
        U = A*U
```
otherwise it's interpreted as `U*A`.

Reply:
Mathematically `A*U` is just as good as `U*A` since everything commute (and in e.g. the  Higham-reference no particular attention is paid to the order of this multiplication). Is there an efficiency advantage to do `*=` for matrices? In my own small tests I saw no performance difference between  `U *= A` and `U=U*A`.

Reply:
> Is there an efficiency advantage to do `*=` for matrices?

No, it's just syntactic sugar and gets expanded. I wasn't paying attention on the background, was just looking at the former code where `A` was multiplied from the left.